inference-engine/ie_bridges/java/cpp/infer_request.cpp

```diff
@@ -0,0 +1,219 @@
+#include <inference_engine.hpp>
+
+#include "openvino_java.hpp"
+#include "enum_mapping.hpp"
+#include "jni_common.hpp"
+
+using namespace InferenceEngine;
+
+static jobject runnable_glob;
+
+JNIEXPORT void JNICALL Java_org_intel_openvino_InferRequest_Infer(JNIEnv *env, jobject obj, jlong addr)
+{
+    static const char method_name[] = "Infer";
+    try
+    {
+        InferRequest *infer_request = (InferRequest *)addr;
+        infer_request->Infer();
+    }
+    catch (const std::exception &e)
+    {
+        throwJavaException(env, &e, method_name);
+    }
+    catch (...)
+    {
+        throwJavaException(env, 0, method_name);
+    }
+}
+
+JNIEXPORT void JNICALL Java_org_intel_openvino_InferRequest_StartAsync(JNIEnv *env, jobject obj, jlong addr)
+{
+    static const char method_name[] = "StartAsync";
+    try
+    {
+        InferRequest *infer_request = (InferRequest *)addr;
+        infer_request->StartAsync();
+    }
+    catch (const std::exception &e)
+    {
+        throwJavaException(env, &e, method_name);
+    }
+    catch (...)
+    {
+        throwJavaException(env, 0, method_name);
+    }
+}
+
+JNIEXPORT jint JNICALL Java_org_intel_openvino_InferRequest_Wait(JNIEnv *env, jobject obj, jlong addr, jint wait_mode)
+{
+    static const char method_name[] = "Wait";
+    try
+    {
+        auto it = wait_mode_map.find(wait_mode_e(wait_mode));
+        InferRequest *infer_request = (InferRequest *)addr;
+
+        if (it == wait_mode_map.end())
+            throw std::runtime_error("No such WaitMode value!");
+
+        InferenceEngine::StatusCode status_code = infer_request->Wait(it->second);
+
+        auto code = status_code_map.find(status_code);
+        if (code == status_code_map.end())
+            throw std::runtime_error("No such StatusCode value!");
+
+        return (jint)(code->second);
+    }
+    catch (const std::exception &e)
+    {
+        throwJavaException(env, &e, method_name);
+    }
+    catch (...)
+    {
+        throwJavaException(env, 0, method_name);
+    }
+}
+
+JNIEXPORT jint JNICALL Java_org_intel_openvino_InferRequest_SetCompletionCallback(JNIEnv *env, jobject, jlong addr, jobject runnable){
+    static const char method_name[] = "SetCompletionCallback";
+    try
+    {
+        InferRequest *infer_request = (InferRequest *)addr;
+        runnable_glob = env->NewGlobalRef(runnable);
```

Comment:
What if we will have multiple networks with async callback? Can we avoid static variable?